File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (nums[0]>target){
            return 0;
        }
        else if (nums.back()<target){
            return nums.size();
        }
        else{
            for (int i= 1; i<nums.size(); i++){
                if (nums[i]==target){
                    return i;
                }
                else if ((nums[i-1]<target) && (nums[i]>target)){
                    return i;
                }
            }
        }
        return 0;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if (nums[0]>target):
            return 0
        elif (nums[-1]<target):
            return len(nums)
        for i in range(len(nums)):
            if nums[i]==target:
                return i
            elif (nums[i-1]<target) and (nums[i]>target):
                    return i